packages/server/src/threads/automation.ts

```diff
@@ -304,41 +292,37 @@ class Orchestrator {
   }
 
   hasErrored(context: AutomationContext): boolean {
-    const [_trigger, ...steps] = context.steps
-    for (const step of steps) {
-      if (step.success === false) {
-        return true
-      }
-    }
-    return false
+    return context._error === true
   }
 
   async execute(): Promise<AutomationResults> {
     return await tracer.trace("execute", async span => {
       span.addTags({ appId: this.appId, automationId: this.automation._id })
 
-      const job = cloneDeep(this.job)
-      delete job.data.event.appId
-      delete job.data.event.metadata
+      const data = cloneDeep(this.job.data)
```

Comment:
This was another big part of why automation tests became slow. `this.job` contains a reference to `InMemoryQueue`, which in turn has a reference to `_messages` which contains all automation queue messages fired during tests. Cloning it in full was expensive.